WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -204,18 +204,20 @@ private async Task StepConnectionConfirmationPhaseAsync(CancellationToken cancel
 					var removedAliceCount = round.Alices.RemoveAll(x => alicesDidNotConfirm.Contains(x));
 					round.LogInfo($"{removedAliceCount} alices removed because they didn't confirm.");
 
-					// Once an input is confirmed and non-zero credentials are issued, it must be included and must provide a
-					// a signature for a valid transaction to be produced, therefore this is the last possible opportunity to
-					// remove any spent inputs.
+					// Once an input is confirmed and non-zero credentials are issued, it is too late to do any
 					if (round.InputCount >= round.Parameters.MinInputCountByRound)
 					{
+						var offendingAliceCounter = 0;
 						await foreach (var offendingAlices in CheckTxoSpendStatusAsync(round, cancel).ConfigureAwait(false))
 						{
-							if (offendingAlices.Any())
-							{
-								var removed = round.Alices.RemoveAll(x => offendingAlices.Contains(x));
-								round.LogInfo($"There were {removed} alices removed because they spent the registered UTXO.");
-							}
+							offendingAliceCounter += offendingAlices.Length;
+						}
+
+						if (offendingAliceCounter > 0)
+						{
+							round.LogInfo($"There were {offendingAliceCounter} alices that spent the registered UTXO. Aborting...");
+							EndRound(round, EndRoundState.AbortedWithError);
```

Comment:
An idea to consider: call `CreateBlameRoundAsync` here? 

Otherwise, an attacker can disrupt rounds just by spending a coin, whitelist in blame round would make this harder. 

In that case do not forget to change here as well https://github.com/molnard/WalletWasabi/blob/master/WalletWasabi/WabiSabi/Client/CoinJoinClient.cs#L292

Reply:
> This is not possible because valid credentials were issued and given to the attacker.

The old credentials can be discarded and new ones issued. This is why I said the connection confirmation phase would be repeated.

Reply:
> An attacker can always disrupt a round by spending a coin, you cannot prevent that unless you pay in advance or something like that.

I didn't say it will prevent it - just makes it harder. 

According to your replies, it is not worth the effort.

Reply:
I think it worth to try. Please take a look again.